src/Sylius/Component/Core/Distributor/MinimumPriceDistributor.php

```diff
@@ -108,4 +107,14 @@ private function exceedsOrderItemMinimumPrice(
     ): bool {
         return $minimumPriceAdjustedByCurrentDiscount >= ($orderItemTotal + $proposedPromotion);
     }
+
+    private function getTotalPrice(OrderItemInterface $orderItem, bool $appliesOnDiscounted, ChannelInterface $channel): int
+    {
+        $variant = $orderItem->getVariant();
+        if ($appliesOnDiscounted === false && !empty($variant->getAppliedPromotionsForChannel($channel))) {
```

Comment:
`!$appilesOnDiscounted`?

Reply:
Also, maybe we could have `hasAppliedPromotionsForChannel($channel): bool` on the variant? It would seem more natural this way